### PR TITLE
Fix linkcheck job

### PR DESCRIPTION
**Pull request by astrofrog**
### Description

This is an attempt to fix the linkcheck job

Fixes https://github.com/astropy/astropy/issues/12382

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/cosmology/dev.rst [hunk 1]
```diff
@@ -234,7 +234,7 @@ integrated within the code structure.
                 ...
 
 For a fully implemented example ``mypackage``, see
-https://github.com/astropy/astropy/tree/main/docs/cosmology/mypackage
+https://github.com/astropy/astropy/tree/main/astropy/cosmology/tests/mypackage
```
Comment (astrofrog) [inline docs/cosmology/dev.rst:237]: @nstarman - is this correct?

Comment (nstarman) [reply docs/cosmology/dev.rst:237]: Yes, thanks! missed that in https://github.com/astropy/astropy/commit/85dc20dc62f60bd17c9b8dbc707d21b30354a207

File: docs/whatsnew/index.rst [hunk 2]
```diff
@@ -13,7 +13,7 @@ functionality highlighted in these pages.
 
    5.1
 
-* `What's New in Astropy 5.0? <https://docs.astropy.org/en/v5.0/whatsnew/5.0.html>`__
+* `What's New in Astropy 5.0? <https://docs.astropy.org/en/v5.0rc1/whatsnew/5.0.html>`__
```
Comment (astrofrog) [inline docs/whatsnew/index.rst:16]: I will update these when 5.0 final is out.

File: astropy/utils/xml/writer.py [hunk 3]
```diff
@@ -154,7 +154,7 @@ def xml_cleaning_method(self, method='escape_xml', **clean_kwargs):
         In order to explicitly allow certain XML tags (e.g. link reference or
         emphasis tags), use ``method='bleach_clean'``.  This sanitizes the data
         string using the ``clean`` function of the
-        `https://bleach.readthedocs.io/en/latest/clean.html <bleach>`_ package.
+        `bleach <https://bleach.readthedocs.io/en/latest/clean.html>`_ package.
```
Comment (pllim) [inline astropy/utils/xml/writer.py:157]: Huh, how did this get past code review?

File: tox.ini [hunk 4]
```diff
@@ -92,12 +92,6 @@ deps =
     devdeps: git+https://github.com/spacetelescope/asdf.git#egg=asdf
     devdeps: git+https://github.com/liberfa/pyerfa.git#egg=pyerfa
 
-    # Linkcheck is broken with cryptic error with Sphinx 3.1.0
-    # https://github.com/astropy/astropy/issues/10489
-    # It also emits a warning from BLS docstring that we cannot banish with numpydoc 1.1.0
-    linkcheck: sphinx<3.1
-    linkcheck: numpydoc<1.1.0
```
Comment (pllim) [inline tox.ini:99]: I knew these will go away if we wait long enough... 😆

**Review by pllim — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**